### ISSUE TITLE
WPT testability

### ISSUE THREAD

**Issue by saschanaz** [w3c/push-api#365]
I'm glad to see some proposals to improve Push API e.g. #360. But I think we need to first establish a good infrastructure to make this API testable and get some shared test cases before adding anything on top of it, because sadly the WPT push-api directory is essentially empty: https://github.com/web-platform-tests/wpt/tree/master/push-api

During TPAC 2023 WebApps session it was briefly mentioned that it could be nice to have some webdriver command for this, and I'd like to go deeper and get an actual achievement in this area.

@beidson, since you proposed #360, has Apple prepared a plan for this?

**Comment by marcoscaceres** [comment]
@saschanaz, we (WebKit) don't have any plans, but it would be nice to figure something out... maybe we could do something with Web Driver, but I've not checked if that's even possible for this case? 

I guess at a minimum, we could add the error checking tests... and if we allow the permission through Web Driver, then it might be possible to somehow send synthetic push messages. 

I think this is something we can do in parallel to #360, but it shouldn't be a precondition to the inclusion of #360... but, #360 should include definitely include whatever tests are possible with the WPT infra we have today.
